src/metabase/util/yaml.clj

```diff
@@ -0,0 +1,47 @@
+(ns metabase.util.yaml
+  (:refer-clojure
+   :exclude
+   [load])
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [metabase.util :as u]
+            [metabase.util.i18n :refer [trs]]
+            [yaml.core :as yaml])
+  (:import java.net.URI
+           [java.nio.file FileSystem FileSystems Path]))
+
+(defmacro with-resource
```

Comment:
We have a macro that does pretty much the same thing here: https://github.com/metabase/metabase/blob/master/src/metabase/plugins/files.clj#L114

If that's useful here too we can move that namespace to something like `metabase.util.files` instead so you don't have to duplicate functionality

Reply:
If my understanding of the underlying Java crap is correct, they aren't the same. `with-open-path-to-resource` doesn't allow you to treat the resource as a directory.

Reply:
It's a `Path` which, if it's a directory, works like you'd expect. We use it to copy all the drivers from either the `resources` directory or the uberjar into `plugins`. See `metabase.plugins`:

```clj
(defn- extract-system-modules! []
  (when (io/resource "modules")
    (let [plugins-path (plugins-dir)]
      (files/with-open-path-to-resource [modules-path "modules"]
        (files/copy-files! modules-path plugins-path)))))
```

Reply:
Oh, cool. I'll reuse that then and move the ns to util.